source/common/upstream/health_checker_event_logger.h

```diff
@@ -2,25 +2,46 @@
 
 #include "envoy/access_log/access_log.h"
 #include "envoy/common/callback.h"
+#include "envoy/config/accesslog/v3/accesslog.pb.h"
+#include "envoy/config/accesslog/v3/accesslog.pb.validate.h"
 #include "envoy/config/core/v3/health_check.pb.h"
 #include "envoy/data/core/v3/health_check_event.pb.h"
 #include "envoy/event/timer.h"
 #include "envoy/runtime/runtime.h"
+#include "envoy/server/factory_context.h"
+#include "envoy/server/health_checker_config.h"
 #include "envoy/stats/scope.h"
 #include "envoy/type/matcher/string.pb.h"
+#include "envoy/upstream/health_check_event_sink.h"
 #include "envoy/upstream/health_checker.h"
 
+#include "source/common/access_log/access_log_impl.h"
 #include "source/common/common/logger.h"
+#include "source/common/config/utility.h"
+#include "source/common/protobuf/utility.h"
 
 namespace Envoy {
 namespace Upstream {
 
 class HealthCheckEventLoggerImpl : public HealthCheckEventLogger {
 public:
   HealthCheckEventLoggerImpl(AccessLog::AccessLogManager& log_manager, TimeSource& time_source,
-                             const std::string& file_name)
-      : time_source_(time_source), file_(log_manager.createAccessLog(Filesystem::FilePathAndType{
-                                       Filesystem::DestinationType::File, file_name})) {}
+                             const envoy::config::core::v3::HealthCheck& health_check_config,
+                             Server::Configuration::HealthCheckerFactoryContext& context)
+      : time_source_(time_source) {
+    if (!health_check_config.event_log_path().empty() /* deprecated */) {
+      file_ = log_manager.createAccessLog(Filesystem::FilePathAndType{
+          Filesystem::DestinationType::File, health_check_config.event_log_path()});
+    }
+
+    for (const auto& config : health_check_config.event_logger()) {
+      if (!config.has_typed_config()) {
+        continue;
```

Comment:
This seems like an error rather than something that should be ignored.
FWIW, I think this should probably be covered by `getAndCheckFactory` that should throw an exception in this case.

Reply:
Correct, it is already handled in the getAndCheckFactory, thanks!